src/App.tsx

```diff
@@ -15,36 +15,104 @@ export const goodsFromServer = [
+
+const sortGoods = (order: SortType, isReversed: boolean): string[] => {
+  let sortedGoods: string[];
+
+  switch (order) {
+    case SortType.ALPHABET:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...preparedGoods];
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(preparedGoods);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.DEFAULT);
+
+  const handleSort = (order: SortType) => {
+    const sortedGoods = sortGoods(order, isReversed);
+
+    setGoods(sortedGoods);
+    setSortOrder(order);
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevState => !prevState);
+  };
+
+  const resetGoods = () => {
+    setGoods(preparedGoods);
+    setSortOrder(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const isResetVisible = sortOrder !== SortType.DEFAULT || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortOrder !== SortType.ALPHABET ? 'is-light' : ''}`}
+          onClick={() => handleSort(SortType.ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${sortOrder !== SortType.LENGTH ? 'is-light' : ''}`}
+          onClick={() => handleSort(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${!isReversed ? 'is-light' : ''}`}
+          onClick={reverseGoods}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isResetVisible && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={resetGoods}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {goods.map(good => (
+          <li key={good} data-cy="Good">
```

Comment:
When rendering a list, you should provide a unique `key` prop to each child. If `good` is not guaranteed to be unique, consider using an index or another unique identifier. Refer to the checklist item 2.